### ISSUE TITLE
Data race in example/simple.go

### ISSUE THREAD

**Issue by HeinOldewage** [gosuri/uiprogress#7]
The data race occurs in `AddBar` and `Listen` on the accessing of `p.Bars`.

Not a big problem, but the example should be clear that the setup should be done before calling `Start`. Alternatively `AddBar` and `Listen` could be made thread safe.

Running `go run -race simple.go` give the following output:

```
C:\Go Programs\src\github.com\gosuri\uiprogress\example\simple>go run -race simple.go
==================
WARNING: DATA RACE
Read by goroutine 7:
  github.com/gosuri/uiprogress.(*Progress).Listen.func1()
      C:/Go programs/src/github.com/gosuri/uiprogress/progress.go:87 +0x89

Previous write by main goroutine:
  main.main()
      C:/Go Programs/src/github.com/gosuri/uiprogress/example/simple/simple.go:11 +0x30e

Goroutine 7 (running) created at:
  github.com/gosuri/uiprogress.(*Progress).Listen()
      C:/Go programs/src/github.com/gosuri/uiprogress/progress.go:92 +0xb7
==================
==================
WARNING: DATA RACE
Read by goroutine 7:
  github.com/gosuri/uiprogress.(*Progress).Listen.func1()
      C:/Go programs/src/github.com/gosuri/uiprogress/progress.go:87 +0xea

Previous write by main goroutine:
  main.main()
      C:/Go Programs/src/github.com/gosuri/uiprogress/example/simple/simple.go:11 +0x2d8

Goroutine 7 (running) created at:
  github.com/gosuri/uiprogress.(*Progress).Listen()
      C:/Go programs/src/github.com/gosuri/uiprogress/progress.go:92 +0xb7
==================
   2s [====================================================================] 100%
Found 2 data race(s)
exit status 66
```

**Comment by gosuri** [comment]
Thank you. Currently, only the [Bar](https://godoc.org/github.com/gosuri/uiprogress#Bar) is tread-safe. Progress needs some serious surgery for thread safely.

**Comment by gosuri** [comment]
Fixed with #11
